pkg/operator/controller/ingress/deployment.go

```diff
@@ -1678,3 +1669,27 @@ func GetMIMETypes(mimeTypes []operatorv1.CompressionMIMEType) []string {
 
 	return mimes
 }
+
+// caps the value of ReloadInterval between the bounds of 1s and 120s
+// returns the default of 5s if the user gives a 0 value
+func capReloadIntervalValue(interval time.Duration) time.Duration {
+	const (
+		maxInterval     = 120 * time.Second
+		minInterval     = 1 * time.Second
+		defaultInterval = 5 * time.Second
+	)
+
+	reloadInterval := defaultInterval
+	if interval == 0 {
+		return reloadInterval
+	}
+
+	if interval < minInterval {
+		reloadInterval = minInterval
+	} else if interval > maxInterval {
+		reloadInterval = maxInterval
+	} else {
+		reloadInterval = interval
+	}
+	return reloadInterval
```

Comment:
```suggestion
	switch interval {
	case interval == 0 :
		return defaultlInterval
	case interval < minInterval :
		return minInterval
	case interval > maxInterval :
		return maxInterval
	default:
		return interval
	}
```

Reply:
How do you use boolean expression (e.g. < and >) with switch case statements?

Reply:
Oops, sorry.  You can just use switch alone:
```suggestion
	switch  {
	case interval == 0 :
		return defaultInterval
	case interval < minInterval :
		return minInterval
	case interval > maxInterval :
		return maxInterval
	default:
		return interval
	}
```

Reply:
And you can get rid of the variable declarations if you do:

```suggestion
	switch  {
	case interval == 0 :
		return 5 * time.Second
	case interval < minInterval :
		return 1 * time.Second
	case interval > maxInterval :
		return 120 * time.Second
	default:
		return 120 * time.Second
	}
```

Reply:
@candita, are you suggesting replacing the named constants with literals?  I prefer using named consts—it should compile to the same code, and the names make the logic clearer.

Reply:
That's fine with me.  I prefer consts too, but not usually within the same function they are declared.